torch/_inductor/codegen/cpp.py

```diff
@@ -1645,8 +1649,8 @@ def _gen_parallel_reduction_buffers(
         if (
             reduction_type == "welford_reduce"
             and welford_weight_reciprocal_vec_fn
-            and hasattr(self, "weight_recp_vec_range")
```

Comment:
Could you please explain why we can't use the original logic?

Reply:
We use `getattr(self, "gen_weight_recps", None)` to check if we need to generate a new weight_recps. If we use the original logic we may generate duplicate definition for weight_recps.